### PR TITLE
[Android] Make sure that ads are enabled when wallet is created

### PR DESCRIPTION
**Pull request by samartnik**
Resolves https://github.com/brave/brave-browser/issues/10526

## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/BraveRewardsPanelPopup.java [hunk 1]
```diff
@@ -137,7 +137,6 @@
     private String currentNotificationId;
     private TextView tvPublisherNotVerified;
     private TextView tvPublisherNotVerifiedSummary;
-    private boolean walletInitialized;          //flag: wallet is initialized
```
Comment (samartnik) [inline android/java/org/chromium/chrome/browser/BraveRewardsPanelPopup.java:140]: This variable is unused

**Comment by samartnik** [comment]
CI failed on audit_deps. Not related to PR.

File: android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayout.java [hunk 2]
```diff
@@ -516,6 +516,8 @@ public void OnWalletInitialized(int error_code) {
         }
       }
     } else if (error_code == BraveRewardsNativeWorker.WALLET_CREATED) { // Wallet created code
+      // Make sure that flag is set as panel can be closed before wallet is created
+      BraveAdsNativeHelper.nativeSetAdsEnabled(Profile.getLastUsedProfile());
```
Comment (NejcZdovc) [inline android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayout.java:520]: @samartnik is this just a front end flag or does this function goes into service and then shared lib to enable ads?

Comment (samartnik) [reply android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayout.java:520]: it makes call `ads_service_->SetEnabled`

Comment (NejcZdovc) [reply android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayout.java:520]: @tmancey I think this should be avoided as UI calls should only be called based on user actions and not based on the observers

**Pull request merged**